core/tests/test_utils.py

```diff
@@ -143,16 +146,44 @@ class TestBase(unittest.TestCase):
     }
 
     def _get_unicode_test_string(self, suffix):
+        """Returns a string for unicode compatibility test.
+
+        Args:
+            suffix: A suffix to append to the unicode string.
+
+        Returns:
+            string. Unicode test string.
+        """
         return '%s%s' % (self.UNICODE_TEST_STRING, suffix)
 
     def setUp(self):
+        """Sets up environment for Test Base
+        (implemented in inherited class).
+
+        Exception:
+            NotImplementedError. The method is not implemented in the 
+            inherited class.
+        """
         raise NotImplementedError
 
     def tearDown(self):
+        """Deactivates environment of Test Base such as logout, delete
+        models and so on (implemented in inherited class).
+
+        Exception:
+            NotImplementedError. The method is not implemented in the 
+            inherited class.
+        """
         raise NotImplementedError
 
     def _assert_validation_error(self, item, error_substring):
-        """Checks that the given item passes default validation."""
+        """Checks that the given item passes default validation.
+
+        Args:
+            item: item to validate.
```

Comment:
Ah, yes, the item here can be various different objects. In that case, write it like this:

`item: *. The item to validate.`

Also, I looked up instances of _assert_validation_error throughout the codebase and it seems to me that there are other _assert_validation_error cases shadowing this one. I suggest removing the more local checks, e.g. the one in core/domain/collection_domain_test.py (please search the codebase for others), and have them use this method directly, in order to avoid confusion.